pvlib/tests/test_irradiance.py

```diff
@@ -207,6 +207,36 @@ def test_haydavies(irrad_data, ephem_data, dni_et):
     assert_allclose(result, [0, 27.1775, 102.9949, 33.1909], atol=1e-4)
 
 
+def test_haydavies_components(irrad_data, ephem_data, dni_et):
+    result = irradiance.haydavies(
+        40, 180, irrad_data['dhi'], irrad_data['dni'], dni_et,
+        ephem_data['apparent_zenith'], ephem_data['azimuth'],
+        return_components=True)
+    result_val = irradiance.haydavies(
+        40, 180, irrad_data['dhi'].values, irrad_data['dni'].values, dni_et,
+        ephem_data['apparent_zenith'].values, ephem_data['azimuth'].values,
+        return_components=True)
+    expected = pd.DataFrame(np.array(
+        [[0, 27.1775, 102.9949, 33.1909],
+         [0, 27.1775, 30.1818, 27.9837],
+         [0, 0, 72.8130, 5.2071],
+         [0, 0, 0, 0]]).T,
+        columns=['sky_diffuse', 'isotropic', 'circumsolar', 'horizon'],
+        index=irrad_data.index
+    )
+    # values
+    assert_allclose(result.sky_diffuse, [0, 27.1775, 102.9949, 33.1909],
+                    atol=1e-4)
+    assert_allclose(result.isotropic, [0, 27.1775, 30.1818, 27.9837],
+                    atol=1e-4)
+    assert_allclose(result.circumsolar, [0, 0, 72.8130, 5.2071], atol=1e-4)
+    assert_allclose(result.horizon, [0, 0, 0, 0], atol=1e-4)
+    assert_frame_equal(result, expected, check_less_precise=4)
+    assert isinstance(result, pd.DataFrame)
```

Comment:
`assert_frame_equal` checks type, values, and more, so I think all the other lines here are actually redundant and can be removed 👍